### PR TITLE
[tsconfig] update to only export a single tsconifg; add declarationMaps

### PR DESCRIPTION
**Pull request by fastfrwrd**
**This change results in a breaking version release (5.0.0)**

I tested out the new configuration and documentation on the folllowing:

- application
  - Babel-only
  - Webpack+ts-loader
  - Webpack+babel-loader
  - Create React App
  - Next.js
- library
  - full TS, tsc-only
  - JS + TS, tsc-only

I removed all of the documentation on babel libraries, as it was written largely for folks transitioning Babel-oriented libraries to TypeScript internally at Spotify, and it is an unwieldy bit of documentation that I don't think we can safely keep up to date long-term.

### CODE REVIEW

File: packages/tsconfig/docs/examples.md [hunk 1]
```diff
@@ -142,70 +133,12 @@ And then in your build steps:
 1. Output JS:
    `tsc`
 1. Output declaration files:
-   `tsc -d --emitDeclarationOnly --allowJs false --declarationDir dist`
+   `tsc -d --emitDeclarationOnly --allowJs false --declarationDir dist --declarationMap`
```
Comment (kirbysayshi) [inline packages/tsconfig/docs/examples.md]: This two-step process is technically not required with the newest version of TS, which can output declaration files while `--allowJs true`!

I wonder if we need these two sections at all. Perhaps the sections should be:

- web-scripts build
- manually configuring tsconfig extends to output a "reasonable default" of es6 + polyfills + allowing JS.

File: packages/tsconfig/docs/examples.md [hunk 2]
```diff
@@ -2,72 +2,45 @@
...
+This setup assumes that Create React App, Next, Webpack, and/or Babel will transpile your bundled assets (aka `tsc` is not used for emitting transpiled code, only for type-checking).
+
+Extend your `tsconfig.json`:
 
-Extend your `tsconfig.json` with our `tsconfig.app.json`
 ```json
 {
-  "extends": "@spotify/tsconfig/tsconfig.app.json",
-  "include": ["src"]
+  "extends": "@spotify/tsconfig"
```
Comment (kirbysayshi) [inline packages/tsconfig/docs/examples.md]: this still needs `include: ["src"]`, doesn't it?

File: packages/tsconfig/package.json [hunk 3]
```diff
@@ -7,8 +7,9 @@
     "type": "git",
     "url": "git+https://github.com/spotify/web-scripts.git"
   },
+  "main": "tsconfig.json",
   "files": [
-    "tsconfig.*.json"
+    "tsconfig*.json"
```
Comment (kirbysayshi) [inline packages/tsconfig/package.json]: Is the `*` necessary?

Comment (josephmcasey) [reply packages/tsconfig/package.json]: Just based off an artless search, I think the `*` is the idiomatic way of adding specification to _which type of typescript configuration the file covers_.

tsconfig

Comment (fastfrwrd) [reply packages/tsconfig/package.json]: @kirbysayshi is right though, we only export the one tsconfig.json now so this future-proofing move isn't necessary. Dropping it for the final PR.

File: packages/tsconfig/docs/examples.md (hunk 2 above)
Comment (fastfrwrd) [reply packages/tsconfig/docs/examples.md]: It's pretty dependent on how you use the TypeScript config - for most of the Webpack-based ways it's not strictly necessary.

File: packages/tsconfig/docs/examples.md (hunk 1 above)
Comment (fastfrwrd) [reply packages/tsconfig/docs/examples.md]: Agreed, simplifies things a lot and avoids outdated configs.

**Pull request merged**